### ISSUE TITLE
Rewrites silently modify query string when "=" is missing

### ISSUE THREAD

**Issue by callumgare** [vercel/next.js#37028]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: #28-Ubuntu SMP Thu Apr 14 04:55:28 UTC 2022
    Binaries:
      Node: 16.15.0
      npm: 8.5.5
      Yarn: 1.22.10
      pnpm: N/A
    Relevant packages:
      next: 12.1.7-canary.8
      react: 17.0.2
      react-dom: 17.0.2

### What browser are you using? (if relevant)

N/A

### How are you deploying your application? (if relevant)

I haven't deployed it yet, currently using the dev server

### Describe the Bug

When a request with query string params that are have only a key (no value or "=") is made to a route which matches an external rewrite Next.js will modify that query string as part of the rewrite to include "=" for each param. For example the query string in "/path?foo=bar&cake&tree" becomes "?foo=bar&cake=&tree=".

I don't know whether sending query string params without a "=" is technically valid but if a client has sent such a request to Next.js I don't think Next.js should silently try to be helpful and modify it to be "valid" since doing so changes the request and that can break things. For example in my use case I'm trying to proxy a request to a third-party api that expects params without "=" and will return an error it has one.

### Expected Behavior

I would expect Next.js to not modify the query string and to pass it on as is (or at the very least for this behavior to be documented).

### To Reproduce

1. Create a new request bin https://requestbin.com/ (If you click the "create a public bin" you don't need to create an account or anything).
2. Create a rewrite in next.config.js with some path as the source and the url of the request bin you created as the destination. E.g.
```javascript
{
    source: '/rewrite-test',
    destination: 'https://en93icipchvct.x.pipedream.net',
}
```
3. Make a request to that route with a query string that includes params that don't have a "=". E.g http://localhost:8000/rewrite-test?foo=bar&cake&tree
4. Observe that the query string is modified in the request send to the request bin.
![Screenshot from 2022-05-19 13-54-38](https://user-images.githubusercontent.com/346340/169206345-0088982b-3400-47ab-a872-75c4163b5bf7.png)

**Comment by icyJoseph** [comment]
Hi,

Not sure it's Next.js doing this, and it might be a standard thing to do...

```js
const params = new URLSearchParams('?foo=bar&cake&tree')
params.get("foo") // 'bar'
params.get("cake") // ''
params.get("tree") //''
params.toString() // 'foo=bar&cake=&tree='
```

**Comment by callumgare** [comment]
If it's not possible to change due to a third-party framework being responsible for this proxing behavior that's fair enough. However I don't believe this is standard practice for proxies. I'm currently using [http-proxy](https://www.npmjs.com/package/http-proxy) which as far as I can tell is the most popular npm module for proxying http requests ([http-proxy-middleware](https://www.npmjs.com/package/http-proxy-middleware) has slightly more weekly downloads but that's actually a wrapper around http-proxy) and I can confirm that it doesn't try to mess with search query strings.

I can understand the behavior of URLSearchParams being more opinionated about the formatting of query strings since it's design goals are different. It's aiming to be a helpful abstraction for building and manipulating search query strings for client-side code as an alternative to the existing practice of just manipulating those strings directly (Node.js also supports it but it was designed by whatwg with browsers in mind). If you're writing browser code and you find yourself needing more control of how the result is formatted then you can just do what was done before URLSearchParams existed, manipulate that the search query string directly, as a string. 

<hr />

BTW for anyone else who wants a workaround we're using an api route with [http-proxy-middleware](https://www.npmjs.com/package/http-proxy-middleware) :
```typescript
/* @flow */
// We currently can't do with Next.js rewrites due to this bug
// https://github.com/vercel/next.js/issues/37028 stopping us
// proxying the google maps api
import { createProxyMiddleware } from "http-proxy-middleware";

const proxyMiddleware = createProxyMiddleware({
    router: req => req.query.origin,
    changeOrigin: true,
    pathRewrite: {
        "^/api/external-resource-proxy/[^/]+/": "/",
    },
});

export default function handler(req: any, res: any): void {
    if (!process.env.NEXT_PUBLIC_SERVER_PROXY) {
        return res.status(400).send("Route not available");
    }
    proxyMiddleware(req, res, (result: any) => {
        if (result instanceof Error) {
            throw result;
        }
    });
}

export const config = {
    api: {
        externalResolver: true,
    },
};
```

**Comment by icyJoseph** [comment]
Right, the thing is that there's no standard behaviour, AFAIK, anyways, https://datatracker.ietf.org/doc/html/rfc3986#section-3.4, the RFC only talks about `key=value` as something people do, but leaves it up to the reader, sorta.

Nice workaround you found! I'll add it to my stash of snippets. 

And yeah, I guess the problem is that Next.js is first a Production Framework for React applications, so there might be a few places where it doesn't behave as entire backend frameworks. 

Still it'd be nice to hear an opinion from the team.
